ruptures/costs/costrank.py

```diff
@@ -97,21 +98,23 @@ def fit(self, signal):
 
         obs, vars = signal.shape
 
-        # argsort gives us 0-based ranks in the range [0, n-1].
-        # Add one to put it in the range [1, n]
-        ranks = np.argsort(signal, axis=0) + 1
+        # Convert signal data into ranks in the range [1, n]
+        ranks = rankdata(signal, axis=0)
         # Center the ranks into the range [-(n+1)/2, (n+1)/2]
         centered_ranks = (ranks - ((obs + 1) / 2)).astype(int)
```

Comment:
Last thing: the centered ranks should not be cast as integers (they might not be)

Reply:
Ah yep, now that we are averaging tied ranks